source/common/version/version.cc

```diff
@@ -40,14 +37,7 @@ const envoy::config::core::v3::BuildVersion& VersionInfo::buildVersion() {
   return *result;
 }
 
-bool VersionInfo::sslFipsCompliant() {
-#ifdef ENVOY_SSL_FIPS
-  RELEASE_ASSERT(FIPS_mode() == 1, "FIPS mode must be enabled in Envoy FIPS configuration.");
```

Comment:
This release-assert was an important test case (because we've messed this up before and didn't know it). Please replace it with something in an integration test.

Reply:
I think the following test already covers this:

https://github.com/envoyproxy/envoy/blob/main/test/common/common/version_test.cc#L39

From what I can tell, this is validating that for Envoy built in FIPS mode, FIPS status is reported correctly (as it's self-reported by boringssl). Is there something else we need to check?

Reply:
Ah, that's probably good enough